src/auth/auth.controller.ts

```diff
@@ -1,25 +1,36 @@
 import type { Request, Response } from "express-serve-static-core";
 import { StatusCodes } from "http-status-codes";
+import { AuthService } from "./auth.service";
+
+const authService = new AuthService();
 
 export default class AuthController {
-  async register(req: Request, res: Response): Promise<void> {
-    res.status(StatusCodes.OK).json({ message: "It works!" });
+  async register(req: Request, res: Response) {
+    const result = await authService.register(req.body);
+    res.status(StatusCodes.CREATED).json(result);
```

Comment:
```suggestion

    res.status(StatusCodes.CREATED).json(result);
```

A little trick to highlight what is actually returned from the function. It really helps to read code faster and decrease cognitive load.